manifests/init.pp

```diff
@@ -1,24 +1,49 @@
-# @summary A short summary of the purpose of this class
+# Manage your next generation Pulp server
 #
-# A description of what this class does
+# === Parameters:
+#
+# $cache_dir::                          Pulp cache directory, default is '/var/lib/pulp/tmp'
+#
+# $config_dir::                         Pulp configuration directory, default is '/etc/pulp'
+#
+# $user::                               Pulp user, default is 'pulp'
+#
+# $group::                              Pulp user group, default is 'pulp'
+#
+# $user_home::                          Pulp user home directory, default is '/var/lib/pulp'
+#
+# $api_host::                           API service host, default is '127.0.0.1'
+#
+# $api_port::                           API service port, default is 24817
+#
+# $content_host::                       Content service host, default is '127.0.0.1'
+#
+# $content_port::                       Content service port, default is 24816
+#
+# $webserver_static_dir::               Directory for Pulp webserver static content, default is '/var/lib/pulp/docroot'
+#
+# $database_name::                      Name of Pulp database, default is 'pulpcore'
+#
+# $database_password::                  Password of Pulp database, default is randomly generated
 #
 # @example
 #   include pulpcore
 class pulpcore (
-  Stdlib::Absolutepath $pulp_cache_dir = '/var/lib/pulpcore/tmp',
-  Stdlib::Absolutepath $pulp_config_dir = '/etc/pulpcore',
-  String $pulp_user = 'pulpcore',
-  String $pulp_group = 'pulpcore',
-  Stdlib::Absolutepath $pulp_user_home = '/var/lib/pulpcore',
-  Stdlib::Host $pulp_api_host = '127.0.0.1',
-  Stdlib::Port $pulp_api_port = 24817,
-  Stdlib::Host $pulp_content_host = '127.0.0.1',
-  Stdlib::Port $pulp_content_port = 24816,
-  Stdlib::Absolutepath $pulp_webserver_static_dir = '/var/lib/pulpcore/docroot',
+  Stdlib::Absolutepath $cache_dir = '/var/lib/pulp/tmp',
+  Stdlib::Absolutepath $config_dir = '/etc/pulp',
+  String $user = 'pulp',
+  String $group = 'pulp',
+  Stdlib::Absolutepath $user_home = '/var/lib/pulp',
+  Stdlib::Host $api_host = '127.0.0.1',
+  Stdlib::Port $api_port = 24817,
+  Stdlib::Host $content_host = '127.0.0.1',
+  Stdlib::Port $content_port = 24816,
+  Stdlib::Absolutepath $webserver_static_dir = '/var/lib/pulp/docroot',
   String $database_name = 'pulpcore',
```

Comment:
I'm wondering if we should prefix it with `postgresql_` as well. I suspect we'll also have redis equivalents. Then `postgresql_database_name` and `redis_database` will be clearer

Reply:
Agreed and done.